code/game/objects/machinery/miner.dm

```diff
@@ -43,21 +50,90 @@
 /obj/machinery/miner/update_icon()
 	switch(miner_status)
 		if(MINER_RUNNING)
-			icon_state = "mining_drill_active"
+			if((mineral_produced == /obj/item/compactorebox/platinum)&&(miner_upgrade_type == MINER_COMPACTOR))
+				icon_state = "mining_drill_active_platinum_[miner_upgrade_type]"
```

Comment:
use istype and add spaces inbetween your boolean operators
(this&&bad)(this && good)